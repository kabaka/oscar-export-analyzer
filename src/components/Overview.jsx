import React from 'react';
import { summarizeUsage, computeAHITrends, computeEPAPTrends } from '../utils/stats';
import MetricGrid from './MetricGrid';
import KPICard from './KPICard';

/**
 * Sparkline chart for a series of numeric data points.
 */
function Sparkline({ data, width = 100, height = 30 }) {
  if (!data || data.length < 2) {
    return null;
  }
  const max = Math.max(...data);
  const min = Math.min(...data);
  const range = max - min || 1;
  const points = data.map((d, i) => {
    const x = (i / (data.length - 1)) * width;
    const y = height - ((d - min) / range) * height;
    return [x, y];
  });
  const pathD = points
    .map(([x, y], idx) => `${idx === 0 ? 'M' : 'L'}${x},${y}`)
    .join(' ');
  return (
    <svg
      className="sparkline"
      viewBox={`0 0 ${width} ${height}`}
      preserveAspectRatio="none"
    >
      <path d={pathD} />
    </svg>
  );
}

/**
 * Overview dashboard displaying high-level KPIs and sparklines.
 */
export default function Overview({ summaryData, clusters, falseNegatives }) {
  const usage = summarizeUsage(summaryData);
  const ahi = computeAHITrends(summaryData);
  const epap = computeEPAPTrends(summaryData);

  return (
    <div className="overview-dashboard">
      <h2 id="overview">Overview Dashboard</h2>
      <MetricGrid>
        <KPICard title="Avg Usage (hrs)" value={usage.avgHours.toFixed(2)}>
          <Sparkline data={usage.usageHours} />
        </KPICard>
        <KPICard title="Median AHI" value={ahi.medianAHI.toFixed(2)}>
          <Sparkline data={ahi.ahis} />
        </KPICard>
        <KPICard title="Median EPAP" value={epap.medianEPAP.toFixed(2)}>
          <Sparkline data={epap.epaps} />
        </KPICard>
        <KPICard title="# Clusters" value={clusters.length.toString()} />
        <KPICard title="# False Negatives" value={falseNegatives.length.toString()} />
      </MetricGrid>
    </div>
  );
}
