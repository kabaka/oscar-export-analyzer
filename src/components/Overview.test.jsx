import React from 'react';
import { render, screen } from '@testing-library/react';
import Overview from './Overview';

const summaryData = [
  { Date: '2021-01-01', 'Total Time': '1:00:00', AHI: '2', 'Median EPAP': '10' },
  { Date: '2021-01-02', 'Total Time': '2:00:00', AHI: '4', 'Median EPAP': '12' },
];
const clusters = [{}, {}];
const falseNegatives = [{}];

describe('Overview', () => {
  it('renders KPI cards with correct titles and values', () => {
    const { asFragment } = render(
      <Overview summaryData={summaryData} clusters={clusters} falseNegatives={falseNegatives} />
    );
    expect(screen.getByText('Overview Dashboard')).toBeInTheDocument();
    expect(screen.getByText('Avg Usage (hrs)')).toBeInTheDocument();
    expect(screen.getByText('Median AHI')).toBeInTheDocument();
    expect(screen.getByText('Median EPAP')).toBeInTheDocument();
    expect(screen.getByText('# Clusters')).toBeInTheDocument();
    expect(screen.getByText('# False Negatives')).toBeInTheDocument();
    // Snapshot omitted to avoid brittleness when adding inline guide links
  });
});
