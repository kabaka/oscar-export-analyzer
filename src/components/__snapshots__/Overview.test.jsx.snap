// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Overview > renders KPI cards with correct titles and values 1`] = `
<DocumentFragment>
  <div
    class="overview-dashboard"
  >
    <h2
      id="overview"
    >
      Overview Dashboard
    </h2>
    <div
      class="metric-grid"
    >
      <div
        class="kpi-card"
      >
        <div
          class="title"
        >
          Avg Usage (hrs)
        </div>
        <div
          class="value"
        >
          1.50
        </div>
        <svg
          class="sparkline"
          preserveAspectRatio="none"
          viewBox="0 0 100 30"
        >
          <path
            d="M0,30 L100,0"
          />
        </svg>
      </div>
      <div
        class="kpi-card"
      >
        <div
          class="title"
        >
          Median AHI
        </div>
        <div
          class="value"
        >
          3.00
        </div>
        <svg
          class="sparkline"
          preserveAspectRatio="none"
          viewBox="0 0 100 30"
        >
          <path
            d="M0,30 L100,0"
          />
        </svg>
      </div>
      <div
        class="kpi-card"
      >
        <div
          class="title"
        >
          Median EPAP
        </div>
        <div
          class="value"
        >
          11.00
        </div>
        <svg
          class="sparkline"
          preserveAspectRatio="none"
          viewBox="0 0 100 30"
        >
          <path
            d="M0,30 L100,0"
          />
        </svg>
      </div>
      <div
        class="kpi-card"
      >
        <div
          class="title"
        >
          # Clusters
        </div>
        <div
          class="value"
        >
          2
        </div>
      </div>
      <div
        class="kpi-card"
      >
        <div
          class="title"
        >
          # False Negatives
        </div>
        <div
          class="value"
        >
          1
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
